.github/workflows/traiage.yaml

```diff
@@ -0,0 +1,151 @@
+          chmod +x "${HOME}/.local/bin/coder"
+          echo "$HOME/.local/bin" >> "${GITHUB_PATH}"
+          "$HOME"/.local/bin/coder version
+          "$HOME"/.local/bin/coder whoami
+
+      - name: Create Coder workspace
+        id: create-workspace
+        env:
+          PREFIX: ${{ inputs.prefix }}
+          CONTEXT_KEY: ${{ steps.extract-context.outputs.context_key }}
+          RUN_ID: ${{ github.run_id }}
+        run: |
+          export WORKSPACE_NAME="${PREFIX}-${CONTEXT_KEY}-${RUN_ID}"
+          echo "Creating workspace: $WORKSPACE_NAME"
+          ./scripts/traiage.sh create
+          echo "workspace_name=$WORKSPACE_NAME" >> "${GITHUB_OUTPUT}"
+          echo "WORKSPACE_NAME=${WORKSPACE_NAME}" >> "${GITHUB_ENV}"
+
+      - name: Send prompt to AI agent inside workspace
+        id: prepare-prompt
+        env:
+          WORKSPACE_NAME: ${{ steps.create-workspace.outputs.workspace_name }}
+          ISSUE_URL: ${{ inputs.issue_url }}
+        run: |
+          PROMPT_FILE=/tmp/prompt.txt
+          trap 'rm -f "${PROMPT_FILE}"' EXIT
+
+          # Fetch issue description using `gh` CLI
+          issue_description=$(gh issue view "${ISSUE_URL}")
+
+          # Write a prompt to PROMPT_FILE
+          cat > "${PROMPT_FILE}" <<EOF
+            Analyze the below GitHub issue description, understand the root cause, and make appropriate changes to resolve the issue.
+
+            ISSUE URL: ${ISSUE_URL}
+            ISSUE DESCRIPTION BELOW:
+
+            ${issue_description}
+          EOF
+
+          echo "WORKSPACE_NAME: ${WORKSPACE_NAME}"
+          PROMPT=$(cat $PROMPT_FILE) ./scripts/traiage.sh prompt
+
+      - name: Commit and push changes
+        id: commit-push
+        if: ${{ inputs.persistence_mode == 'push' }}
+        run: |
+          echo "Committing and pushing changes in workspace: $WORKSPACE_NAME"
+          ./scripts/traiage.sh commit-push
+
+      - name: Create and upload archive
+        id: create-archive
+        if: ${{ inputs.persistence_mode == 'archive' }}
+        env:
+          DESTINATION_PREFIX: ${{ secrets.TRAIAGE_DESTINATION_PREFIX }}
+        run: |
+          echo "Creating archive for workspace: $WORKSPACE_NAME"
+          ./scripts/traiage.sh archive
+          echo "archive_url=${DESTINATION_PREFIX%%/}/$WORKSPACE_NAME.tar.gz" >> "${GITHUB_OUTPUT}"
+
+      - name: Report results
+        env:
+          ISSUE_URL: ${{ inputs.issue_url }}
+          CONTEXT_KEY: ${{ steps.extract-context.outputs.context_key }}
+          WORKSPACE_NAME: ${{ steps.create-workspace.outputs.workspace_name }}
+          ARCHIVE_URL: ${{ steps.create-archive.outputs.archive_url }}
+        run: |
+          {
+            echo "## TrAIage Results";
+            echo "- **Issue URL:** ${ISSUE_URL}";
+            echo "- **Context Key:** ${CONTEXT_KEY}";
+            echo "- **Workspace:** ${WORKSPACE_NAME}";
+            echo "- **Archive URL:** ${ARCHIVE_URL}"
+          } >> "${GITHUB_STEP_SUMMARY}"
```

Comment:
Could we also output here how to utilize the artifact?

PS. What's the difference of throwing it into step summary vs just outputting to stdout?

Reply:
Regarding the "how-to" -- I think we will want to add some level of automation around this in the associated template. Something like the below:

```
data "coder_parameter" "resume_context_key" {
  name = "resume_context_key"
  type = "string"
  default = ""
}

locals {
  archive_url = "gs://${var.gcp_bucket}/traiage/${context_key}"
}

resource "coder_script" "fetch_context" { ... }

```